### ISSUE TITLE
[top_darjeeling] fake pinmux or direct connection

### ISSUE THREAD

**Issue by tjaychen** [lowRISC/opentitan#17085]
### desc

The integrated subsystem has needs for external interrupts / wake ups / strap like pins.

We need to decide whether to make these dedicated top level ports that forward directly to the modules of concern, or make use of a "fake" pinmux to keep mostly aligned with discrete behavior.

Using wakeups as an example, we could route the external wakeups to `pwrmgr`, but we could also route them to a `virtual pinmux which still has much of the wake up pulse / level detection logic that could be useful.

For interrupts, see #17070.

For straps, it is a similar story wakeups.  We could connect them directly to `cio_gpio`, or go through a `virtual pinmux` to keep largely the same structure.  

A virtual pinmux would likely be all "dedicated" pins to make this more straightforward.

**Comment by tjaychen** [comment]
@msfschaffner 
@a-will

**Comment by msfschaffner** [comment]
External IRQs, wakeups and reset requests have been implemented via a `soc_proxy` module (similar to `sensor_ctrl`) which can expose internal connections to the SoC via the intersignal topgen feature.

For straps, we decided to add a snapshot feature to the GPIO which is triggered by the pwrmgr strap output (in the same way as the pinmux strap sampling). With GPIOs exposed as dedicated IOs, the GPIO can act as a bank of up to 32bit of HW assisted straps that are sampled at power-up.

CC @andreaskurth
